core/server/master/src/main/java/alluxio/master/metastore/rocks/RocksBlockStore.java

```diff
@@ -161,18 +157,28 @@ public long size() {
   @Override
   public void close() {
     mSize.reset();
+    LOG.info("Closing RocksBlockStore and recycling all RocksDB JNI objects");
     mRocksStore.close();
+    mIteratorOption.close();
+    mDisableWAL.close();
+    mCfOpts.close();
+    // TODO(jiacheng): not final, can't set to null
+    // mDisableWAL = null;
+    LOG.info("RocksBlockStore closed");
   }
 
   @Override
   public List<BlockLocation> getLocations(long id) {
     byte[] startKey = RocksUtils.toByteArray(id, 0);
     byte[] endKey = RocksUtils.toByteArray(id, Long.MAX_VALUE);
 
-    // Explicitly hold a reference to the ReadOptions object from the discussion in
+    // Explicitly hold a reference to the objects from the discussion in
     // https://groups.google.com/g/rocksdb/c/PwapmWwyBbc/m/ecl7oW3AAgAJ
-    final ReadOptions readOptions = new ReadOptions().setIterateUpperBound(new Slice(endKey));
-    try (RocksIterator iter = db().newIterator(mBlockLocationsColumn.get(), readOptions)) {
+    try (
+         // TODO(jiacheng): why final?
+         final Slice slice = new Slice(endKey);
+         final ReadOptions readOptions = new ReadOptions().setIterateUpperBound(slice);
+         final RocksIterator iter = db().newIterator(mBlockLocationsColumn.get(), readOptions)) {
```

Comment:
According to the Oracle doc, the close order is opposite to the declaration order
https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html

Reply:
javadoc your clarification here?

Reply:
I believe you mean comment